source/helpers/render-semantic-release-commit-types.ts

```diff
@@ -0,0 +1,28 @@
+import zipTextNodes from 'zip-text-nodes';
+
+// Using https://github.com/angular/angular/blob/main/CONTRIBUTING.md?rgh-link-date=2024-09-09T05%3A51%3A31Z#-commit-message-format as the reference for the seamntic commit message format.
+
+export function getSemanticCommitAndScope(commitTitle: string): [string, string?] | undefined {
+	const match = /^(\w*)(?:\((.+)\))?!?:.*/.exec(commitTitle);
+
+	if (!match) {
+		return undefined;
+	}
+
+	return [match[1], match[2]];
+}
+
+export function strip(
+	element: HTMLElement,
+	from: string,
+): void {
+	const modified = element.textContent
+		.replace(from, match => `<s>${match}</s>`);
+
+	const temporaryDiv = document.createElement('div');
+	temporaryDiv.innerHTML = modified;
```

Comment:
As for performance, I agree it’s not the most well-optimized code for this intent, but it’s certainly tested and fully safe. In fact this XSS doesn’t actually make it to the page because the zip-text-nodes package will throw an error.

After merge I’ll check if I can move this strip function into zip-text-node to use its internal utilities directly, without the zipping part (but it’s unlikely, it would be more complex)

Reply:
The alternative solution is the one I had earlier, where I was iterating the nodes and contacting the strings until I saw ":" and stripped it away. No XSS & dependencies and easier to understand than an abstract concept like zipping nodes. I'll leave the decision to you.

Reply:
https://gist.github.com/marcojahn/482410b728c31b221b70ea6d2c433f0c see this

Reply:
Also it works with the regex I have locally:

![image](https://github.com/user-attachments/assets/e053b2d8-8c8c-4c38-9e28-be739f781b99)

Consider the link above though.

Reply:
Question (I’m on mobile): do commit titles with emojis have extra elements in them? GitHub usually uses `<g-emoji>` to render them. If not, we might be able to simplify this a lot.

Reply:
Seems to be just an emoji ![image](https://github.com/user-attachments/assets/7c038791-15c4-42a1-a11b-7c557f833e19)

The regex \w does not match an emoji though: 

![image](https://github.com/user-attachments/assets/f00678ae-ebca-4534-a591-9722496fa855)